connect/runtime/src/test/java/org/apache/kafka/connect/integration/ErrantRecordSinkConnector.java

```diff
@@ -44,6 +49,12 @@ public ErrantRecordSinkTask() {
         public void start(Map<String, String> props) {
             super.start(props);
             reporter = context.errantRecordReporter();
+            executorService = Executors.newSingleThreadExecutor();
+        }
+
+        @Override
+        public void stop() {
+            ThreadUtils.shutdownExecutorServiceQuietly(executorService, 4, TimeUnit.SECONDS);
```

Comment:
Just curious--any rationale for four seconds specifically?

Reply:
I picked this because it was less than the default graceful shutdown timeout, but i don't think it's particularly sensitive.